laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -581,270 +433,89 @@ private function insertOption()
-            if (!is_null($schedule_data2)) {
-                Option::where("date", $date)
-                    ->where("resident", $resident)
-                    ->where("option", $choice)
-                    ->update([
-                        "schedule" => $split[1],
-                        "attending" => $attending,
-                        "milestones" => $_REQUEST["milestones2"],
-                        "objectives" => $_REQUEST["objectives2"],
-                        "anesthesiologist_id" => $pref_anest2,
-                        "isValid" => 1,
-                    ]);
-            } else {
-                Option::where("date", $date)
-                    ->where("resident", $resident)
-                    ->where("option", $choice)
-                    ->delete();
-            }
-        } else {
-            $overwrittenChoices[1] = 0;
-            if (!is_null($schedule_data2)) {
-                // Insert data
-                Option::insert([
-                    "date" => $date,
-                    "resident" => $resident,
-                    "schedule" => $split[1],
-                    "attending" => $attending,
-                    "option" => $choice,
-                    "milestones" => $_REQUEST["milestones2"],
-                    "objectives" => $_REQUEST["objectives2"],
-                    "anesthesiologist_id" => $pref_anest2,
-                    "isValid" => 1,
-                ]);
-            }
-        }
-
-        //Third choice data
-        $choice++;
-        $schedule_data3 = null;
-        $attending = null;
-        // Insert third choice data if it exists
-        if ($split[2] != 0) {
-            $schedule_data3 = ScheduleData::where("id", $split[2])->get();
-            // Get attending id
-            $attending_string = $schedule_data3[0]["lead_surgeon"];
-            $attending = substr(
-                $attending_string,
-                strpos($attending_string, "[") + 1,
-                strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-            );
-        }
-
-        if (isset($_REQUEST["pref_anest3"])) {
-            if ($_REQUEST["pref_anest3"] != 0) {
-                $pref_anest3 = $_REQUEST["pref_anest3"];
-            }
-        }
+        for ($i = 0; $i < 3; $i++) {
+            $choice = $i + 1;
+            if ($split[$i]) {
+                $pref_anest[$i] = null;
+                $schedule_data[$i] = ScheduleData::where("id", $split[$i])->get();
+                $attending_string = $schedule_data[$i][0]["lead_surgeon"];
+                $date = $schedule_data[$i][0]["date"];
+                $attending = substr(
+                    // get rid of [number] after the surgeon name
+                    $attending_string,
+                    strpos($attending_string, "[") + 1,
+                    strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
+                );
```

Comment:
I'm pretty sure at least some of this is repeated from the `updateMilestones` method.

Seems like a private method might reduce repetition.

Reply:
With the new regex being just 2 lines of code, I'm going to leave this as is